app/lib/methods/sendFileMessage.js

```diff
@@ -55,57 +48,77 @@ export async function sendFileMessage(rid, fileInfo, tmid) {
 			}
 		});
 
-		const result = await _ufsCreate.call(this, fileInfo);
-
-		promises[fileInfo.path] = RNFetchBlob.fetch('POST', result.url, {
-			'Content-Type': 'octet-stream'
-		}, data);
-		// Workaround for https://github.com/joltup/rn-fetch-blob/issues/96
-		setTimeout(() => {
-			if (promises[fileInfo.path] && promises[fileInfo.path].uploadProgress) {
-				promises[fileInfo.path].uploadProgress((loaded, total) => {
-					database.write(() => {
-						fileInfo.progress = Math.floor((loaded / total) * 100);
-						try {
-							database.create('uploads', fileInfo, true);
-						} catch (e) {
-							return log('err_send_file_message_create_upload_2', e);
-						}
-					});
-				});
-			}
-		});
-		await promises[fileInfo.path];
-
-		const completeResult = await _ufsComplete.call(this, result.fileId, fileInfo.store, result.token);
-
-		await _sendFileMessage.call(this, completeResult.rid, {
-			_id: completeResult._id,
-			type: completeResult.type,
-			size: completeResult.size,
-			name: completeResult.name,
-			description: completeResult.description,
-			url: completeResult.path
-		}, {
-			tmid
-		});
+		XHR[fileInfo.path] = xhr;
+		xhr.open('POST', uploadUrl);
 
-		database.write(() => {
-			const upload = database.objects('uploads').filtered('path = $0', fileInfo.path);
-			try {
-				database.delete(upload);
-			} catch (e) {
-				log('err_send_file_message_delete_upload', e);
-			}
+		formData.append('file', {
+			uri: fileInfo.path,
+			type: fileInfo.type,
+			name: fileInfo.name
 		});
-	} catch (e) {
-		database.write(() => {
-			fileInfo.error = true;
-			try {
-				database.create('uploads', fileInfo, true);
-			} catch (err) {
-				log('err_send_file_message_create_upload_3', err);
+		formData.append('description', fileInfo.description);
+
+		if (tmid) {
+			formData.append('tmid', tmid);
+		}
+
+		xhr.setRequestHeader('X-Auth-Token', token);
+		xhr.setRequestHeader('X-User-Id', id);
+
+		const handleProgress = ({ total, loaded }) => {
+			database.write(() => {
+				fileInfo.progress = Math.floor((loaded / total) * 100);
+				try {
+					database.create('uploads', fileInfo, true);
+				} catch (e) {
+					return log('err_send_file_message_create_upload_2', e);
+				}
+			});
+		};
+
+		xhr.addEventListener('progress', handleProgress);
```

Comment:
This doesn't work.
You should use `xhr.upload.onprogress`.
Example:
```jsx
xhr.upload.onprogress = (event) => {
	console.log('progress: ' + event.loaded + ' / ' + event.total);
}
```

Reference: https://github.com/facebook/react-native/pull/17312

Reply:
done